OpenRA.Mods.Common/Widgets/Logic/ServerCreationLogic.cs

```diff
@@ -182,8 +184,23 @@ void BuildNotices()
 			}
 		}
 
+		void UpdatePreview(string uid)
+		{
+			if (modData.MapCache[uid].Status != MapStatus.Available)
+				preview = modData.MapCache[modData.MapCache.ChooseInitialMap(Game.Settings.Server.Map, Game.CosmeticRandom)];
+			else
+				preview = modData.MapCache[uid];
+		}
+
 		void CreateAndJoin()
 		{
+			// Refresh MapCache
+			if (modData.MapCache[preview.Uid].Status != MapStatus.Available)
+			{
+				UpdatePreview(preview.Uid);
+				return;
```

Comment:
Would it be a problem to not `return` here? It looks a bit weird when you click "Create" and just the map changes or refreshes. Since we're not directly starting the game but just opening a lobby, the map changing in the worst case shouldn't be a problem.

Reply:
What do you think the alternative should be? I've implemented it this way for all other menus, to me it makes sense from a ux perspective. The game responds to your file changes without doing anything unexpected

Reply:
I haven't thought about this specific PR in any detail, but these comments make me thing that we may want a mechanism for the ChromeLogic to be able to bind a listener to a map changed event so it can refresh the map automatically and only when required.

Reply:
> I haven't thought about this specific PR in any detail, but these comments make me think that we may want a mechanism for the ChromeLogic to be able to bind a listener to a map changed event so it can refresh the map automatically and only when required.

I had noticed that the lobby menu already refreshes the map every tick, just without updating MapCache. It is not true for other menus, of which we have just a few. Server creation and Mission Selection, both are easily solved by placing the refresh on the start button

Reply:
> What do you think the alternative should be?

The alternative would be starting the updated map, which is as I said/assume what the user would want anyway.

Reply:
> The alternative would be starting the updated map, which is as I said/assume what the user would want anyway.

As I noted on discord I am 100% against this. I don't want the UI to force undesired behaviour onto the user. If a player picks a map, that is then deleted. I would much rather have the map refresh on `Start Game` rather than force a lobby with a random map. This also feels intuitive, cause the user has manually edited his map folder, so a refresh won't be a surprise. Later on when we support transient notifications in all menu's we could add this here as well.

While this is a rather insignificant detail due to the minute chance of this actually be triggered, I'm really not a fan of setting a president for UI that forces undesired outcomes onto the player.